C0T0/Classes/ApiDataSource/URLRequestBuilder.swift

```diff
@@ -10,20 +10,41 @@ protocol URLRequestBuilder {
 }
 
 class StandardURLRequestBuilder: URLRequestBuilder {
-
+    private enum Constants {
+        static let http = "http"
+        static let https = "https"
+        static let slashes = "://"
+        static let httpProtocol = Constants.http + Constants.slashes
+        static let httpsProtocol = Constants.https + Constants.slashes
+    }
+    
     private let host: String
-
+    private let scheme: String
     init(host: String) {
+        var host = host
+        var scheme = ""
+        if host.contains(Constants.httpProtocol) {
+            host = host.replacingOccurrences(of:Constants.httpProtocol , with: "")
+            scheme = Constants.http
+        } else if host.contains(Constants.httpsProtocol) {
+            host = host.replacingOccurrences(of: Constants.httpsProtocol, with: "")
+            scheme = Constants.https
+        }
         self.host = host
+        self.scheme = scheme
     }
 
     func build(from request: ApiRequest) -> URLRequest? {
-        guard let  url = URL(string: host + request.endpoint) else { return nil }
-        var urlRequest = URLRequest(url: url)
+        var components = URLComponents()
```

Comment:
![image](https://user-images.githubusercontent.com/1502999/45618082-d528e580-ba74-11e8-8af0-9198da30d9a7.png)
As you can see on the screenshot, `URLComponents` can split address into scheme and host, so you don't need to do it manually, just pass whole `host` to it :)